### PR TITLE
 update bstats and git-commit-id-plugin dependencies

### PR DESCRIPTION
**Pull request by MajdT51**
- remove bstats maven dependancy and use class instead
- update pl.project13.maven to io.github.git-commit-id
- update org.apache.commons to 3.15.0
- add log for critical exception in read file

### CODE REVIEW

**Comment by RoboMWM** [comment]
why not use maven repo? I thought using the maven repo and shading is the recommended approach?

**Comment by MajdT51** [comment]
> why not use maven repo? I thought using the maven repo and shading is the recommended approach?

I saw many plugins using the class instead of the dependancy (including GP). Therefore, I assumed using the class is more efficient and lighter (jar file is a little bit smaller now). From your experience which one do You recommend :D?

**Comment by RoboMWM** [comment]
I recommend having it shaded in maven as it's generally easier to maintain for updates.

Having it separate in GP is in part for 2 reasons:
1) I needed to modify the plugin name it sends, since I'm differentiating between the "legacy" and "new" GP. The only way to do this is via modifying the class.
2) I'd like to have deliberate control over how dependencies are managed in GP.

It's up to you if you want to do it because of reason 2 (or any other reason). It's not more efficient as it's the same code being run. I'm not sure if it's shading other classes, you may get some file space gains with the `minimizeJar` option. In my opinion, it's an additional maintenance cost to maintain/update with bstats, as well as not being very clear on what version of the class it is. If these tradeoffs are worth the benefits of extra control, then that's fine to do this.

**Comment by MajdT51** [comment]
thank you, I will change it back and only update io.github.git-commit-id in this branch

**Comment by MajdT51** [comment]
@RoboMWM  It can be merged now if you okay with it

File: pom.xml [hunk 1]
```diff
@@ -58,6 +58,7 @@
                     <artifactSet>
                         <includes>
                             <include>org.bstats:bstats-bukkit</include>
+                            <include>org.bstats:bstats-base</include>
```
Comment (RoboMWM) [inline pom.xml:61]: interesting, is this now required?

File: src/me/ryanhamshire/PopulationDensity/PopulationDensity.java [hunk 2]
```diff
@@ -535,7 +536,7 @@ public void onEnable()
         try
         {
             Metrics metrics = new Metrics(this, 3343);
-            metrics.addCustomChart(new Metrics.SimplePie("bukkit_impl", this::getVersionImplementation));
+            metrics.addCustomChart(new SimplePie("bukkit_impl", this::getVersionImplementation));
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/PopulationDensity.java:539]: I guess this is ok, kind of a style thing I guess.

Comment (MajdT51) [reply src/me/ryanhamshire/PopulationDensity/PopulationDensity.java:539]: no, in the newest version SimplePie is not a static Class of Metrics anymore. It is now part of org.bstats.charts

File: pom.xml (hunk 1 above)
Comment (MajdT51) [reply pom.xml:61]: org.bstats.charts needed for SimplePie which is not included in bstats-bukkit anymore. It is now in https://github.com/Bastian/bstats-metrics/tree/v3.0.2/base/src/main/java/org/bstats/charts

File: src/me/ryanhamshire/PopulationDensity/PopulationDensity.java (hunk 2 above)
Comment (MajdT51) [reply src/me/ryanhamshire/PopulationDensity/PopulationDensity.java:539]: new Version:

image

-----

old version

image

Comment (RoboMWM) [reply src/me/ryanhamshire/PopulationDensity/PopulationDensity.java:539]: Ah, good to know thank you!

**Pull request merged**